pkg/apis/sources/v1alpha1/cloudevents_lifecycle.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/triggermesh/triggermesh/pkg/apis/common/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// GetGroupVersionKind implements kmeta.OwnerRefable.
+func (*CloudEventsSource) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("CloudEventsSource")
+}
+
+// GetConditionSet implements duckv1.KRShaped.
+func (*CloudEventsSource) GetConditionSet() apis.ConditionSet {
+	return v1alpha1.DefaultConditionSet
+}
+
+// GetStatus implements duckv1.KRShaped.
+func (s *CloudEventsSource) GetStatus() *duckv1.Status {
+	return &s.Status.Status
+}
+
+// GetSink implements EventSource.
+func (s *CloudEventsSource) GetSink() *duckv1.Destination {
+	return &s.Spec.Sink
+}
+
+// GetStatusManager implements EventSource.
+func (s *CloudEventsSource) GetStatusManager() *v1alpha1.StatusManager {
+	return &v1alpha1.StatusManager{
+		ConditionSet: s.GetConditionSet(),
+		Status:       &s.Status,
+	}
+}
+
+// AsEventSource implements EventSource.
+func (s *CloudEventsSource) AsEventSource() string {
+	return ""
```

Comment:
Instead of returning empty values, let's just remove this method and the one below. Implementing `EventSource` is nice but only optional.